File: Java/BrokenGreetings.java
/**
Description: The sign up problem :)
Correct this code, so that the greet function returns the expected value.
*/

public class Person {
    String name;

    public Person(String personName) {
        name = personName;
    }

    public String greet(String yourName) {
        return String.format("Hi %s, my name is %s", yourName, name);
    }
}

File: JavaScript/BrokenGreetings.js
/**
Description: The sign up problem :D
Correct this code, so that the greet function returns the expected value.
*/

function Person(name){
  this.name = name;
}

Person.prototype.greet = function(otherName){
  return "Hi " + otherName + ", my name is " + this.name;
}